datastore/postgres/updatevulnerabilities.go

```diff
@@ -181,18 +210,20 @@ func (s *MatcherStore) updateVulnerabilities(ctx context.Context, updater string
 		}
 
 		if len(oldVulns) > 0 {
-			for _, v := range vulns {
+			vulnIter(func(v *claircore.Vulnerability, _ error) bool {
```

Comment:
should this and `delIter` make assumptions about the passed in `err` that it's safe to ignore?